# Generated by Django 2.2 on 2019-04-12 03:47

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('codex', '0015_auto_20190327_1520'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='EncounterInstance',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=256)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
                ('owner_id', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='CreatureInstance',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('currentHitPoints', models.IntegerField(null=True)),
                ('maxHitPoits', models.IntegerField(null=True)),
                ('ac', models.IntegerField(null=True)),
                ('creature', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='codex.Creature')),
                ('encounter', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, to='battle.EncounterInstance')),
            ],
        ),
    ]
